app/models/ext_management_system.rb

```diff
@@ -173,6 +174,42 @@ def hostname_format_valid?
 
   default_value_for :enabled, true
 
+  after_save :change_maintenance_for_child_managers, :if => proc { |ems| ems.changed_attributes.include?('enabled') }
+
+  def disable!
+    _log.info("Disabling EMS [#{name}] id [#{id}].")
+    update!(:enabled => false)
+    _log.info("Disabling EMS [#{name}] id [#{id}] successful.")
+  end
+
+  def enable!
+    _log.info("Enabling EMS [#{name}] id [#{id}].")
+    update!(:enabled => true)
+    _log.info("Enabling EMS [#{name}] id [#{id}] successful.")
+  end
+
+  # Move ems to maintenance zone and backup current one
+  def pause!
+    _log.info("Pausing EMS [#{name}] id [#{id}].")
+    update!(
+      :backup_zone => zone,
+      :zone        => Zone.maintenance_zone,
```

Comment:
@Fryguy maybe there can be nil? It simplifies it and I don't see validation on zone's presence, right?

Reply:
https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/cloud_manager.rb#L39

Reply:
also whole UI doesn't think that zone == nil is possible, I don't want to rewrite whole UI plugin :)

Reply:
zone being nil is bad because the MiqQueue thinks that zone=nil is equivalent to "all zones".  So work would go on the  queue with the zone value set to nil, and then get picked up anywhere in the region.  clean slate, I wish nil was never used for "all" in the MiqQueue, and instead we used "*" or something.  [Billion Dollar Mistake](https://www.lucidchart.com/techblog/2015/08/31/the-worst-mistake-of-computer-science/) :wink: